src/main/kotlin/blackjack/controller/Controller.kt

```diff
@@ -0,0 +1,98 @@
+package blackjack.controller
+
+import blackjack.model.CardDeck
+import blackjack.model.Dealer
+import blackjack.model.ErrorMessage
+import blackjack.model.Player
+import blackjack.model.Players
+import blackjack.model.ResultEvaluator
+import blackjack.view.InputView
+import blackjack.view.OutputView
+import blackjack.view.OutputView.displayInitialCardsMessage
+
+class Controller {
+    private val deck = CardDeck()
+    private var _players: Players? = null
+    private val dealer = Dealer()
+
+    private val players: Players
+            get() = _players!!
+
+    fun runGame() {
+        _players = initializePlayers()
+        OutputView.displayPlayerNames(players)
+        drawInitialCards()
+        OutputView.displayFirstCardMessage(dealer)
+        players.forEach { OutputView.displayAllCardsMessage(it) }
+        players.dealCards()
+        dealer.dealCards()
+        displayCardsAndTotal()
+        printResults()
+    }
+
+    fun printResults() {
+        val evaluator = ResultEvaluator(players, dealer)
+        val results = evaluator.calculateResults()
+        OutputView.displayResults(results)
+    }
+
+    fun displayCardsAndTotal() {
+        OutputView.displayParticipantStatus(dealer)
+        players.forEach { OutputView.displayParticipantStatus(it) }
+    }
+
+    fun Dealer.dealCards() {
+        while (this.shouldDraw()) {
+            OutputView.displayDealerDrawMessage()
+            this.drawCard(deck)
+        }
+    }
+
+    fun Players.dealCards() {
+        players.forEach { dealCards(it) }
+    }
```

Comment:
Keeping these extension functions in between controller methods will make maintaining the project very difficult, since we're extending domain's logic in controller. Why don't we move these to somewhere else?

I do understand this happened because we want to print the status in these methods, but we'll have to take out `OutputView` logic from these domain logic to properly implement MVC pattern.